code-review/class.User.php

```diff
@@ -0,0 +1,316 @@
+		$sql = "SELECT * FROM users WHERE id = :id LIMIT 1";
+		$params = array(
+				':id' => $id
+			       );
+		$result_array=self::findBySQL($sql, $params);
+		return !empty($result_array)?array_shift($result_array):false;
+	}
+
+	private static function findBySQL($sql, $params = NULL) {
+		global $db;
+		$result_set=$db->query($sql, $params);
+		$object_array=array();
+		while($row=$db->fetchArray($result_set)) {
+			$object_array[]=self::instantiate($row);
+		}
+		return $object_array;
+	}
+
+	public static function addUser($username=null, $full_name=null, $email=null, $password=null,
+			$joined=null, $is_activated=null, $type=null, $token=0) {
+		global $db;
+		$password = Utils::hash($password);
+
+		$params = array(
+				':username' => $username,
+				':full_name' => $full_name,
+				':email' => $email,
+				':password' => $password,
+				':joined' => $joined,
+				':token' => $token
+			       );
+		if($is_activated!=null && $type!=null){
+			$params[':is_activated'] = $is_activated;
+			$params[':type'] = $type;
+			$sql = "INSERT INTO users (username, full_name, email, password, joined, is_activated, type, token)";
+			$sql .= " VALUES (:username, :full_name, :email, :password, :joined, :is_activated, :type, :token)";
+		} else {
+			$sql = "INSERT INTO users (username, full_name, email, password, joined, token)";
+			$sql .= "VALUES (:username, :full_name, :email, :password, :joined, :token)";
+		}
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	public static function addToken($username, $token){
+		global $db;
+		$sql = "UPDATE users SET token=:token WHERE username = :username";
+		$params = array(
+				':token' => $token,
+				':username' => $username
+			       );
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	public static function updatePassword($password, $username){
+		global $db;
+		$password = Utils::hash($password);
+		$sql="UPDATE users SET password=:password, token = 0 WHERE username = :username";
+		$params = array(
+				':password' => $password,
+				':username' => $username
+			       );
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+
+	public static function updateLastVisit($username){
+		global $db;
+		$last_visit = date("Y-m-d H-i-s");
+		$sql = "UPDATE users SET last_visit = :last_visit WHERE username = :username";
+		$params = array(
+				':last_visit' => $last_visit,
+				':username' => $username
+			       );
+		$query = $db->query($sql, $params);
+		if ($db->affectedRows($query)) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+
+	public static function getNumberOfUsers($search=null,$category=null) {
+		global $db;
+		if ($search != null && $category != null) {
+			$params[':search_string'] = '%'.$search.'%';
+			switch ($category) {
```

Comment:
### Pažeidžiamos taisyklės
- [ ] Nenaudoti switch sakinių be default dalies
### Komentarai
Reikėtų aprašyti `default case`, klaidoms valdyti